assets/js/modules/analytics-4/utils/withCustomDimensions.js

```diff
@@ -0,0 +1,374 @@
+					// custom dimensions.
+					if (
+						customDimensionsCreationErrors.some(
+							isInsufficientPermissionsError
+						)
+					) {
+						return (
+							<MetricTileError
+								title={ __(
+									'Insufficient permissions',
+									'google-site-kit'
+								) }
+								{ ...commonErrorProps }
+							>
+								<div className="googlesitekit-report-error-actions">
+									<span className="googlesitekit-error-retry-text">
+										{ createInterpolateElement(
+											__(
+												'Permissions updated? <a>Retry</a>',
+												'google-site-kit'
+											),
+											{
+												a: (
+													<Link
+														onClick={
+															handleCreateCustomDimensions
+														}
+													/>
+												),
+											}
+										) }
+									</span>
+									<span className="googlesitekit-error-retry-text">
+										{ createInterpolateElement(
+											__(
+												'You’ll need to contact your administrator. <a>Learn more</a>',
+												'google-site-kit'
+											),
+											{
+												a: (
+													<Link
+														href={ helpLink }
+														external
+													/>
+												),
+											}
+										) }
+									</span>
+								</div>
+							</MetricTileError>
+						);
+					}
+
+					// Handle generic errors encountered while creating
+					// custom dimensions.
+					return (
+						<MetricTileError
+							title={ __(
+								'Analytics update failed',
+								'google-site-kit'
+							) }
+							{ ...commonErrorProps }
+						>
+							<div className="googlesitekit-report-error-actions">
+								<Button
+									onClick={ handleCreateCustomDimensions }
+								>
+									{ __( 'Retry', 'google-site-kit' ) }
+								</Button>
+								<span className="googlesitekit-error-retry-text">
+									{ createInterpolateElement(
+										__(
+											'Retry didn’t work? <a>Learn more</a>',
+											'google-site-kit'
+										),
+										{
+											a: (
+												<Link
+													href={ helpLink }
+													external
+												/>
+											),
+										}
+									) }
+								</span>
+							</div>
+						</MetricTileError>
+					);
+				}
+
+				if ( ! hasCustomDimensions ) {
+					return (
+						<MetricTileError
+							title={ __( 'No data to show', 'google-site-kit' ) }
+							{ ...commonErrorProps }
+						>
+							<div className="googlesitekit-report-error-actions">
+								<Button
+									onClick={ handleCreateCustomDimensions }
+								>
+									{ __( 'Update', 'google-site-kit' ) }
+								</Button>
+								<span className="googlesitekit-error-retry-text">
+									{ __(
+										'Update Analytics to track metric',
+										'google-site-kit'
+									) }
+								</span>
+							</div>
+						</MetricTileError>
+					);
+				}
+			}
+
+			return <WrappedComponent { ...props } />;
+		};
+
+		return WithCustomDimensionsComponent;
```

Comment:
We also need to give this HOC a name in the same way we do it for `WhenActive`: https://github.com/google/site-kit-wp/blob/8aff60e144bf845cb787ce23f79c96b15a6206cd/assets/js/util/when-active.js#L90-L97

Reply:
Updated, thanks!